lalrpop/src/normalize/token_check/mod.rs

```diff
@@ -131,17 +165,49 @@ impl<'grammar> Validator<'grammar> {
             // If there is no extern token definition, then collect
             // the terminal literals ("class", r"[a-z]+") into a set.
             None => match term {
-                TerminalString::Bare(c) => {
-                    // Bare identifiers like `x` can never be resolved
-                    // as terminals unless there is a conversion
-                    // defined for them that indicates they are a
-                    // terminal; otherwise it's just an unresolved
-                    // identifier.
-                    panic!("bare literal `{}` without extern token definition", c);
-                }
-                TerminalString::Literal(l) => {
-                    self.all_literals.entry(l).or_insert(span);
-                }
+                // FIMXE: Should not allow undefined literals if no CatchAll
+                TerminalString::Bare(c) => match self.match_mappings {
+                    Some(ref m) => {
```

Comment:
Good place for a comment, to remind the reader what the `Some` branch means. i.e.:

```
// If the user gave us a `match` mapping, try to map this back to the actual
// regular expression.
//
// Example:
//     match { r"foo" => "bar" }
// If the user gave us `"bar"`, we would map back to `r"foo"`
```

(something like that, I actually haven't read into what this code does yet;)